src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -170,22 +180,25 @@ public virtual void Abort()
 
                                     _cachedPingMessage = Protocol.WriteToArray(PingMessage.Instance);
 
-                                    Log.UsingHubProtocol(_logger, Protocol.Name);
-
                                     UserIdentifier = userIdProvider.GetUserId(this);
 
                                     if (Features.Get<IConnectionInherentKeepAliveFeature>() == null)
                                     {
                                         // Only register KeepAlive after protocol negotiated otherwise KeepAliveTick could try to write without having a ProtocolReaderWriter
-                                        Features.Get<IConnectionHeartbeatFeature>()?.OnHeartbeat(state => ((HubConnectionContext)state).KeepAliveTick(), this);
+                                        Features.Get<IConnectionHeartbeatFeature>()?.OnHeartbeat(state => ((HubConnectionContext) state).KeepAliveTick(), this);
                                     }
 
+                                    Log.NegotiateComplete(_logger, Protocol.Name);
+                                    await WriteNegotiateResponseAsync(new NegotiationResponseMessage(null));
```

Comment:
null?

Reply:
For the error. Because negotiate succeeds here then it is null. Want a static helper method to create NegotiateResponseMessage with error defaulting to null?